doc/how-to-run-in-windows.rst

```diff
@@ -0,0 +1,352 @@
+``libssl.a``:
+
+::
+
+   > ghc ... -IC:\\msys64\\ucrt64\\includes -LC:\\msys64\\ucrt64\\lib -lssl
+
+This should result in a succesful compilation.
+
+Using UNIX-like binaries with GHC
+---------------------------------
+
+As said above, the binaries that we might need to use are (by default) located
+in two places: ``C:\msys64\usr\bin`` and ``C:\msys64\<environment>\bin``. The
+easiest way to give GHC access to these executables is by including those
+directories in your ``PATH``.
+
+- On PowerShell:
+  ::
+
+     PS> $env:Path += ';C:\msys64\usr\bin;C:\msys64\ucrt64\bin'
+
+     PS> (Get-Command make.exe).Path
+     C:\msys64\usr\bin\make.exe
+
+     PS> make --version
+     GNU Make 4.4.1
+     Built for x86_64-pc-msys
+     ...
+
+- On ``cmd``:
+  ::
+
+     cmd> set PATH=%PATH%;C:\msys64\usr\bin;C:\msys64\ucrt64\bin
+
+     cmd> where make
+     C:\msys64\usr\bin\make.exe
+
+     cmd> make --version
+     GNU Make 4.4.1
+     Built for x86_64-pc-msys
+     ...
+
+- On MSYS2 this is usually already set (otherwise your whole system is probably
+  broken), but just in case one would do the following:
+
+  ::
+
+     msys2> export PATH=$PATH:/c/msys64/usr/bin:/c/msys64/ucrt64/bin
+
+     msys2> which make
+     /usr/bin/make
+
+     msys2> make --version
+     GNU Make 4.4.1
+     Built for x86_64-pc-msys
+     ...
+
+If one really wants, it is possible to edit the system ``PATH`` to add these two
+directories (although as explained below, Cabal will automatically add them for
+us) by doing:
+
+1. Open the search bar and search for "Edit the system environment variables"
+
+2. Switch to the "Advanced" tab and click on "Environment Variables..."
+
+3. Double click on ``PATH`` and add ``;<dir1>;<dir2>`` at the end of the already existing value
+
+Once these directories are in the path, any UNIX-like utilities we might need
+will be in scope, so configuring packages should be able to find them.
+
+Using Cabal on Windows
+----------------------
+
+Cabal will also be installed automatically by GHCup. Once the location of the
+binary is present in the ``PATH``, ``cabal`` can be invoked from any Windows
+shell, just as GHC.
+
+::
+
+   > cabal --help
+   Command line interface to the Haskell Cabal infrastructure.
+
+   See http://www.haskell.org/cabal/ for more information.
+
+   Usage: cabal-3.10.2.1.exe [GLOBAL FLAGS] [COMMAND [FLAGS]]
+
+   ...
+
+As both ``cabal.exe`` and ``ghc.exe`` are placed in the same location by GHCup,
+normally cabal will be able to invoke GHC as said location will be in the
+``PATH``.
+
+All the above extra configurations will also be needed when using ``cabal``, and
+one can provide all of them via the following local project file:
+
+::
+
+   msys2> cat << EOF > cabal.project.local
+   package *
+     extra-include-dirs: C:\msys64\ucrt64\include
+     extra-lib-dirs: C:\msys64\ucrt64\lib
+     extra-prog-path: C:\msys64\ucrt64\bin,
+                      C:\msys64\usr\bin
+   EOF

```

Comment:
I read this section and think "ok, this sounds like it's going to solve my problem" and then immediately find out that it's not the best way to solve my problem, because the guide then talks about setting up global configuration. I'd consider putting this after the best solution or omitting it.